firebase.android.js

```diff
@@ -15,13 +15,28 @@ var fbCallbackManager = null;
 var GOOGLE_SIGNIN_INTENT_ID = 123;
 var REQUEST_INVITE_INTENT_ID = 48
 
-var gson = typeof(com.google.gson) === "undefined" ? null : new com.google.gson.Gson();
+function lazy(action) {
+    var _value;
+    var that = this;
+
+    return function() {
+        return that._value || (that._value = action());
+    }
+}
+
+var gson = lazy(function() {
```

Comment:
Turned out I had to add `new` here (and 4 lines below) otherwise the cached values would interfere.